### PR TITLE
feat: ESLint and Prettier

### PR DESCRIPTION
**Pull request by NickyMeuleman**
wip

### CODE REVIEW

**Comment by NickyMeuleman** [comment]
Please review those @typescript rules that are disabled.

Having trouble with `parserOptions.project` path. When I enter `./tsconfig.json` the eslint in the project works fine, but the eslint extensions freaks out, resulting in an error at the top of every file.
When I enter `./packages/frontend/tsconfig.json` the editor extension pick it up no problem, but the eslint in the project (called through CLI) throws not found errors

**Comment by NickyMeuleman** [comment]
Using this PR as a place to write down issues I encounter while working this.

Lots of `import/no-unresolved` issues (100+)
think it is related to:
https://github.com/benmosher/eslint-plugin-import/issues/720

Will try to update package, hope it doesn't break airbnb ruleset.

**Comment by NickyMeuleman** [comment]
imports still not working, trying to work in a `.eslintrc.js` file is hard. I miss my intellisense :p

Will wait until that PR gets fleshed out and merged. Then I'll try again.	

Laptop gets supertoasty while reloading the project. 100% CPU/RAM/HDD usage does that, I guess.
Sleep now, it's 1 am

**Comment by NickyMeuleman** [comment]
Well I tried again before it was merged anyway.
Imports now work 🎉 	
Turns out I made a stupid error and also included their plugin configuration in `extends`, whoops!

Making many commits here so I can use this to keep track of what's done.
Don't worry, I'll squash and merge when I'm done.

**Comment by NickyMeuleman** [comment]
Done with frontend!
It's wonderful, WONDERFUL

Tomorrow is backend and then extracting shared stuff to the project root.

NB: only files affected by the `tsconfig` in frontend are affected right now, as defined by the `project` key in `.eslintrc`. This is needed for type specific linting 🎉🚀

**Comment by NickyMeuleman** [comment]
Backend was easy.

Ran into trouble when trying to add ESLint to the entire workspace (to later remove duplicates from both frontend and backend)

Seems like our `.js` files are not being linted

https://github.com/typescript-eslint/typescript-eslint/issues/109

**Comment by NickyMeuleman** [comment]
Tried to cantalize the eslint dependencies.

Now imports are broken again (eventhough the rules are still local)
Running ESLint results in lots of `Resolve error: typescript with invalid interface loaded as resolver`

**Comment by NickyMeuleman** [comment]
Having a base config in root and others in packages could be a problem
https://github.com/eslint/eslint/issues/820

The way I'm seeing it working right now is that rules I placed in root (because both frontend and backend use them) are simply not being picked up.

**Comment by NickyMeuleman** [comment]
Well, monorepos confuse me.

Did further centralization of rules (according to eslint docs)
The Typescript resolver plugin was not installed in root somehow.
That coupled with a `settings` key solved import problems.

`no-unused-vars` started firing in backend when I centralized that rule, weird, `@typescript-eslint/no-unused-vars` replaced it.

Running `yarn lint:ws` still doesn't work. It hangs a while, then outputs this error:
code_2019-03-07_17-39-53


All in all, ready for review. This monorepo configuration is mindbending to me so I might be making some obvious error.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/node_modules/*
```
Comment (pantharshit00) [inline .eslintignore]: Delete this file

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const path = require("path");
```
Comment (pantharshit00) [inline .eslintrc.js]: delete this one as well.

**Comment by NickyMeuleman** [comment]
Ok.
Going to merge this now.

Only linting inside `packages/frontend` or `packages/backend`

Linting ´.js´ files acted weird in testing. Not being picked up by the CLI tool. Because the `--ext` option is passed for `.ts` and `.tsx`. It refused to run when you pass `.js` to that `--ext` option.
Also see https://github.com/typescript-eslint/typescript-eslint/issues/109

**Pull request merged**